### DISCUSSION TITLE
Next 13 unable to fetch data server-side and render component client side simultaneously (as it did previously in v12).

### DISCUSSION THREAD

**Discussion by App-Index** [vercel/next.js#42318] [Help]
Previously you could write a single component where

getServerSideProps() is called on the server, 
data is passed as a prop to the component, which runs client side, where you can use things like useState() as usual.

This is no longer possible?

If you use a client side component, you are no longer calling your database (Securely) on the server side using getServerSideProps().
If you don't use a client component, you cannot use useState() and other client side hooks.

Am I missing an important feature?

**Comment by r-martins** [comment]
getServerSideProps can only be used on pages, not components.

From https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props
> If you export a function called getServerSideProps (Server-Side Rendering) **from a page**, ...
